[hotfix] improvements to savepoint docs

knaufk opened the pull request:
## What is the purpose of the change

* minor improvements to savepoint docs

## Brief change log

see commits

dawidwys commented on docs/content/docs/ops/state/savepoints.md:
```diff
@@ -199,23 +200,20 @@ This submits a job and specifies a savepoint to resume from. You may give a path
-There is no one perfect restore mode, each comes with its own price, but they can serve different
-purposes. Nevertheless, we believe the default *NO_CLAIM* mode is a good tradeoff for most usages,
-as it provides clear ownership with a small price for the first checkpoint after the restore.
+Each restore mode serves a specific purposes. 
```
`purposes` -> `purpose`

dawidwys commented on docs/content/docs/ops/state/savepoints.md:
```diff
@@ -229,40 +227,31 @@ user's control and never delete any of the files. In this mode you can start mul
-The other available mode is the *CLAIM* mode. In this mode Flink will claim the ownership of the snapshot
-and might delete it at a certain point in time, once it becomes subsumed. Flink keeps around
+The other available mode is the *CLAIM* mode. In this mode Flink claims ownership of the snapshot
+and essentially treats it like a checkpoint: its controls the lifecycle and might delete it if it is not needed for recovery anymore.
```
`its controls` -> `it controls`? `controls its lifecycle`?

dawidwys commented on docs/content/docs/ops/state/savepoints.md:
```diff
@@ -229,40 +227,31 @@ user's control and never delete any of the files. In this mode you can start mul
 </div>
 
-{{< hint warning >}}
+{{< hint info >}}
```
tbh, I'd rather keep it as a `warning` as those are limitations, but I am not strong on that

dawidwys reviewed (approved):
Thanks for the improvements. LGTM

dawidwys commented:
One more comment. Just to say it out loud.

I think using contractions i.e. don't should be fine. After all, it is not that formal ;)

dawidwys commented:
@knaufk shall we proceed with that?

knaufk commented:
@dawidwys Thank you for the review. I addressed your comments. I once read that contractions should in general not be used in written language. I don't have a strong opinion.
